static/app/views/dashboards/exportDashboard.tsx

```diff
@@ -41,65 +49,117 @@ function getAPIParams(structure) {
-        }
+function normalizeData(
+  source: DashboardDetails
+): Omit<DashboardDetails, (typeof deleteProperties)[number]> {
+  const payload: Omit<DashboardDetails, (typeof deleteProperties)[number]> = {
+    title: '',
+    filters: {},
+    projects: [],
+    widgets: [],
+    environment: [],
+  };
 
-        if (typeof source[data][attr] !== 'string') {
-          if (last_data_attr === null) {
-            last_data_attr = data;
-          }
+  for (const property in payload) {
+    if (property in source) {
+      let data: any[] = [];
 
-          if (last_data_attr !== null) {
-            payload[last_data_attr].push(normalize_data(source[data][attr]));
-            data = String(last_data_attr);
-            last_data_attr = null;
-          }
-        } else {
-          payload[data].push(source[data][attr]);
-        }
+      // if there is a nested object with properties that should be deleted
+      if (['widgets', 'filters'].includes(property)) {
+        // get the object properties so that we can loop through them
+        const type = getType(property);
+        data = normalizeNestedObject(source[property], type);
+      } else {
+        data = source[property];
       }
-    } else {
-      if (!delete_properties.includes(data)) {
-        if (data === 'limit' && source[data] === null) {
-          payload[data] = 5;
+
+      payload[property] = data;
+    }
+  }
+
+  return payload;
+}
+
+function normalizeNestedObject(object, structure) {
+  const nestedObjectArray: any[] = [];
+  const nestedObject = structure;
+
+  for (const index in object) {
+    for (const property in structure) {
+      if (property in object[index]) {
+        let data: any[] = [];
+
+        if (['queries'].includes(property)) {
+          // get the object properties so that we can loop through them
+          const type = getType(property);
+          data = normalizeNestedObject(object[index][property], type);
         } else {
-          payload[data] = source[data];
+          data = object[index][property];
         }
+
+        nestedObject[property] = data;
       }
     }
+
+    nestedObjectArray.push(nestedObject);
   }
 
-  return payload;
+  return nestedObjectArray;
 }
 
-function cleanTitle(title) {
-  const regex = /[^a-z0-9]/gi;
-  const formatted_title = title.replace(regex, '-');
-  const date = new Date();
-  return `${formatted_title}-${date.toISOString()}`;
+function getType(property) {
+  let structure = {};
+
+  switch (property) {
+    case 'widgets':
+      structure = {
+        title: '',
+        description: '',
+        interval: '',
+        queries: [],
+        displayType: '',
+      };
+      break;
+    case 'filters':
+      structure = {
+        release: '',
+      };
+      break;
+    case 'queries':
+      structure = {
+        aggregates: [],
+        columns: [],
+        conditions: [],
+        name: '',
+        orderby: '',
+        fieldAliases: [],
+        fields: [],
```

Comment:
I tried to use the type definitions here (similar to what was done on line 55) but I did not know how to declare an empty object for the `displayType` of the Widget Type.

If I were to do
`var widgetStructure : Omit<Widget, (typeof deleteProperties)[number]> = {
        displayType: ,
        title: "",
        description: "",
        interval: "",
        queries: [],
      }`

`displayType: {}` gives me an error, I guess its because it is an `enum` instead of `type`